Compiler/script2/cc_compiledscript.cpp

```diff
@@ -84,19 +82,22 @@ AGS::ccCompiledScript::~ccCompiledScript()
 }
 
 // [fw] Note: Existing callers expected this function to return < 0 on overflow
-AGS::GlobalLoc AGS::ccCompiledScript::AddGlobal(size_t value_size, void *value_ptr)
+AGS::GlobalLoc AGS::ccCompiledScript::AddGlobal(size_t const value_size, void *value_ptr)
 {
     if (0u == value_size)
         return globaldatasize; // nothing to do
 
     // The new global variable will be moved to &(globaldata[offset])
     size_t const offset = (globaldatasize < 0) ? 0u : static_cast<size_t>(globaldatasize);
 
-    void *chunk_start = globaldata;
+    void *globaldata_chunk = globaldata;
+    size_t const el_size = sizeof(globaldata[0u]); // size of 1 element, i.e. 1 char
+    size_t const needed = offset + value_size;
+    size_t const minimum_allocated = 100u;
```

Comment:
what's the purpose of these hardcoded numbers ? shouldnt we allocate just as many bytes as we need ?